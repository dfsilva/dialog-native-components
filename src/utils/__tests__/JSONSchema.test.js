/*
 * Copyright 2017 dialog LLC <info@dlg.im>
 */

import { parseJSONSchema } from '../JSONSchema';

describe('parseJSONSchema', () => {
  const errors = [
    null,
    {},
    {
      properties: { foo: null }
    },
    {
      properties: {
        foo: {
          type: 'email',
          title: ''
        }
      }
    },
    {
      properties: {
        foo: {
          type: 'string',
          title: 1
        }
      }
    }
  ];

  errors.forEach((it) => {
    const input = JSON.stringify(it);
    test(`should be invalid "${input}"`, () => {
      expect(() => parseJSONSchema(input)).toThrowErrorMatchingSnapshot();
    });
  });

  const ok = [
    {
      properties: {
        foo: { type: 'string', title: 'foo' }
      }
    },
    {
      properties: {
        foo: { type: 'number', title: 'foo' }
      }
    },
    {
      properties: {
        foo: { type: 'integer', title: 'foo' }
      }
    },
    {
      properties: {
        foo: { type: 'boolean', title: 'foo' }
      }
    }
  ];

  ok.forEach((it) => {
    const input = JSON.stringify(it);
    test(`should be valid "${input}"`, () => {
      expect(parseJSONSchema(input)).toEqual(it);
    });
  });
});
