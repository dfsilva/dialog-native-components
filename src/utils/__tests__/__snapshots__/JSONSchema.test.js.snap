// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseJSONSchema should be invalid "{"properties":{"foo":{"type":"email","title":""}}}" 1`] = `"Schema property \\"foo\\" has unsupported type"`;

exports[`parseJSONSchema should be invalid "{"properties":{"foo":{"type":"string","title":1}}}" 1`] = `"Schema property \\"foo\\" title is not a string\\""`;

exports[`parseJSONSchema should be invalid "{"properties":{"foo":null}}" 1`] = `"Schema property \\"foo\\" is not a plain object"`;

exports[`parseJSONSchema should be invalid "{}" 1`] = `"Schema properties is not a plain object"`;

exports[`parseJSONSchema should be invalid "null" 1`] = `"Schema is not a plain object"`;
